smoke_test/hevc_fei_smoke_test.py

```diff
@@ -0,0 +1,1258 @@
+            LOG.write('size after sample_fei'.ljust(25) + ' : ' +
+                      str(round(float(self.size_test)/2**20, 3)) + 'MB\n')
+
+            diff_size = round(100 * float(abs(self.size_ref - self.size_test)) /
+                              min(self.size_ref, self.size_test), 5)
+            if abs(self.size_ref - self.size_test) <= 0.2*min(self.size_ref, self.size_test):
+                LOG.write('Diff: 0 <= ' + str(diff_size) + '% <= 20% - ok\n')
+            else:
+                LOG.write('Diff: ' + str(diff_size) + '% < 0 or ' + str(diff_size) +
+                          '% > 20% - fail\n')
+
+
+class TestCaseBitExact(TestCase):
+    is_bit_exact = False
+
+    def run(self, case_id):
+        TestCase.run(self, case_id)
+        if self.err_code != 0:
+            return 1
+        self.is_bit_exact = filecmp.cmp(PATH_TO_IO + '/' + str(case_id).rjust(4, '0') + '.hevc',
+                                        PATH_TO_IO + '/' + str(case_id).rjust(4, '0') + '.cmp')
+        if self.is_bit_exact:
+            return 0
+        return 1
+
+    def write_details(self):
+        TestCase.write_details(self)
+        if self.err_code == 0:
+            LOG.write('\n---------VERIFICATION---------\n\nBit to bit comparing:\n')
+
+            if self.is_bit_exact:
+                LOG.write('PASS\n')
+            else:
+                LOG.write('FAILED\n')
+
+
+class RunnableBinary:
+    path_to_bin = ''
+    params = ''
+    log_content = ''
+    return_code = 0
+
+    def __init__(self, binary, params):
+        self.path_to_bin = binary
+        self.params = params
+
+
+def silent_file_remove(filename):
+    try:
+        os.remove(filename)
+    except OSError as err:  # this would be "except OSError, e:"
+        if err.errno != errno.ENOENT:  # errno.ENOENT = no such file or directory
+            raise  # re-raise exception if a different error occured
+
+
+# check binaries and streams
+if not os.access(ASG, os.X_OK):
+    print('No ASG or it cannot be executed')
+    sys.exit(ReturnCode.ERROR_ACCESS_DENIED.value)
+if not os.access(FEI_EXTRACTOR, os.X_OK):
+    print('No hevc_fei_extractor or it cannot be executed')
+    sys.exit(ReturnCode.ERROR_ACCESS_DENIED.value)
+if not os.access(SAMPLE_FEI, os.X_OK):
+    print('No sample_hevc_fei or it cannot be executed')
+    sys.exit(ReturnCode.ERROR_ACCESS_DENIED.value)
+if not os.access(SAMPLE_ENCODE, os.X_OK):
+    print('No sample_encode or it cannot be executed')
+    sys.exit(ReturnCode.ERROR_ACCESS_DENIED.value)
+if not os.access(MFX_PLAYER, os.X_OK):
+    print('No mfx_player or it cannot be executed')
+    sys.exit(ReturnCode.ERROR_ACCESS_DENIED.value)
+if not os.access(os.path.normpath(PATH_TEST_STREAM), os.F_OK):
+    print(os.path.normpath(PATH_TEST_STREAM) + " doesn't exist")
+    sys.exit(ReturnCode.ERROR_ACCESS_DENIED.value)
+
+# test cases
+TEST_CASES = []
+
+# case counter is used for case numbering
```

Comment:
**One of the main things:**
I am sure that all this repeated structure (with cases) below can be easily changed to the dynamic creation with using of one good structured dict.

Data should manage the flow of the program.